admin/workflow_wiki/README.md

```diff
@@ -0,0 +1,445 @@
+
+1. Create a new branch to work off of.
+    * When we create a new branch, we want to make sure our new branch is branched off of the latest version of `upstream/main`.
+      Thus, we first checkout `main`.
+      ```console
+      $ git checkout main
+      ```
+      Then we `pull` the latest changes.
+      ```console
+      $ git pull upstream main
+      ```
+      Now your local branch `main` is up to date with `upstream/main`.
+    * Now create your new branch. By default it will be based off of the branch you currently have checked out (`main` in this case which is now the same as `upstream/main`).
+      Name this new branch something related to the changes you're making.
+      ```console
+      $ git checkout -b <your branch name>
+      ```
+
+2. Make code changes.
+    * Checkout the local branch related to this change.
+      ```console
+      $ git checkout <your branch name>
+      ```
+    * Make sure you are working off of the most up to date code in case something changes that affects my changes.
+      ```console
+      $ git pull --rebase upstream main
+      ```
+    * Implement your code changes.
+    * Test your code changes.
+    * Stage your changes.
+      ```console
+      $ git add <list of files/directories you changed>
+      ```
+    * Commit your changes with a useful commit message.
+      For the first commit you make, you should format your commit message in the following way:
+      ```console
+      [team/project] commit summary
+
+      PR_link_keyword #issue_number
+
+      Detailed commit message
+      ```
+
+      Here is a good example:
+      ```console
+      [nav] Refactor navigation simulator
+
+      resolves umrover #308
+
+      Refactor navigation simulator to use Vue and TypeScript.
+      This refactor is indented to make the simulator easier to maintain,
+      make it easier to add new features, and move towards using
+      technologies that are more commonly used in industry.
+
+      We are keeping around the old version of the simulator
+      temporarily in case the refactored version has any major
+      problems. This should be deleted when the correctness of the
+      refactored version is confirmed.
+      ```
+      Don't worry about this _too much_ because you will have the chance to change later.
+    * Use `git status` and `git log` often!
+      ```console
+      $ git status
+      ```
+      ```console
+      $ git log
+      ```
+    * Push your changes to your remote repository.
+      ```console
+      $ git push origin <your branch name>
+      ```
+    * Repeat this step as many times as necessary until your have met all the requirements laid out in the description of the issue you are working on.
+
+3. Testing code changes.
+    * Perform full system testing and integration testing to verify your changes and verify no regression to other changes as a result of your changes.
```

Comment:
Is full system testing making sure all of the systems just build properly or also run properly? Could we apply multiple PR's to the same branch and test them all in one test? If we need runtime testing I'm pretty sure auton would require people from multiple subteams and to assess whether their code stack has been affected. I'm not sure how realistic this is assuming we do one Integration Test a week where we'd have to test the PR's from three different subteams individually as well as collectively. I feel like unit testing each PR on a subteam basis and doing a collective integration test for all incoming PR's would be more realistic. I think that if you don't relax this requirement a bit people are just going to make large PR's so they can do fewer tests, which I think defeats the purpose of this workflow.

Reply:
No. That is what Travis does. This is making sure that stuff still runs properly. You are correct that this wording is a bit ambiguous in that I don't spell out the testing required. I think the right answer is a middle ground between the extremes you laid out. There does need to be some verifying that we are not breaking things but I do not expect that you gather someone from every team. On the flip side, you *do* need to verify that you have not broken your interface with another team. I will try to clarify this.

I understand your concern with requiring strict testing however the solution is not to just skip testing. We have spent countless hours in the past debugging problems from interfaces that were broken that could be solved by this. While your proposal saves time in the short run, it is my belief that this saves time in the long run.

Reply:
Seeing as you have more experience I will trust your judgement. I think specifically checking for violations of the interfaces during integration testing is more realistic and feasible

Reply:
I agree. I like the modification you brought about! This is also something we will work to get better at as a team, because, as you pointed out, it is not something we have great infrastructure for at the moment.

Reply:
Are we going to mention travis in this section then?

Reply:
We definitely can if you think it's best. However, you can't merge without passing travis anyway so I figured it's be too not add details here if we didn't need to. The travis details will come into play in the second half. What is your preference?

Reply:
Okay if travis comes in later it should be fine